cppcheck-suppressions-list.txt

```diff
@@ -0,0 +1,2 @@
+missingInclude
+useStlAlgorithm
```

Comment:
Aw. I like STL algorithms!

Reply:
There is just 1 (one) 'problem' cppcheck finds:

```cxx
list<string> getNodeIds( list<xmlNode*> nodes ) {
  list<string> ids;
  for (auto& node : nodes)
  {
    ids.push_back(TiCC::getAttribute(node, "id"));
  }
  return ids;
}
```

I could change it into this:

```cxx
list<string> getNodeIds( list<xmlNode *> nodes ) {
  list<string> ids;
  transform(
    begin(nodes),
    end(nodes),
    back_inserter(ids),
    []( xmlNode const *node ) {
      return TiCC::getAttribute( node, "id" );
    } );
  return ids;
}
```
But I don't like it.

Reply:
I totally agree:
For code readability, I would also prefer the first  variant.
And i doubt if the second variant gives any significant advantage in speed too.

(note: maybe the auto variable in variant 1 can be const?)

Reply:
In all fairness, a thorough STL variant would go like this:

```cpp
list<string> getNodeIds( list<xmlNode *> nodes ) {
  list<string> ids;
  transform(
    begin(nodes), end(nodes), back_inserter(ids),
    bind(TiCC:getAttribute, _1, "id") );
  return ids;
}
```

That said, I don't expect any performance difference, either (especially not with compiler optimizations turned on), and it is mostly a matter of taste, though avoiding hand-written loops (or lambdas) does reduce the surface where errors can occur a bit.

Reply:
Note to the above: you need a line going

```cpp
using namespace std::placeholders;
```

somewhere near the top of the translation unit in order for the `_1` notation to work.